crates/astria-composer/tests/blackbox/composer.rs

```diff
@@ -98,6 +174,54 @@ async fn mount_broadcast_tx_sync_mock(
     );
     Mock::given(matcher)
         .respond_with(ResponseTemplate::new(200).set_body_json(&jsonrpc_rsp))
+        .up_to_n_times(1)
+        .expect(1)
+        .mount_as_scoped(server)
+        .await
+}
+
+/// Deserizalizes the bytes contained in a `tx_sync::Request` to a signed sequencer transaction and
+/// verifies that the contained sequence action is for the given `expected_chain_id`. It then
+/// rejects the transaction for an invalid nonce.
+async fn mount_broadcast_tx_sync_invalid_nonce_mock(
+    server: &MockServer,
+    expected_chain_id: &'static str,
+) -> MockGuard {
+    use proto::{
+        generated::sequencer::v1alpha1 as raw,
+        native::sequencer::v1alpha1::SignedTransaction,
+        Message as _,
+    };
+    let matcher = move |request: &Request| {
+        let wrapped_tx_sync_req: request::Wrapper<tx_sync::Request> =
+            serde_json::from_slice(&request.body)
+                .expect("can't deserialize to JSONRPC wrapped tx_sync::Request");
+        let raw_signed_tx = raw::SignedTransaction::decode(&*wrapped_tx_sync_req.params().tx)
+            .expect("can't deserialize signed sequencer tx from broadcast jsonrpc request");
+        let signed_tx = SignedTransaction::try_from_raw(raw_signed_tx)
+            .expect("can't convert raw signed tx to checked signed tx");
+        debug!(?signed_tx, "sequencer mock received signed transaction");
+        let Some(sent_action) = signed_tx.actions().get(0) else {
+            panic!("received transaction contained no actions");
+        };
+        let Some(sequence_action) = sent_action.as_sequence() else {
+            panic!("mocked sequencer expected a sequence action");
+        };
+        sequence_action.chain_id == expected_chain_id.as_bytes()
+    };
+    let jsonrpc_rsp = response::Wrapper::new_with_id(
+        Id::Num(1),
+        Some(tx_sync::Response {
+            code: AbciCode::INVALID_NONCE.into(),
+            data: vec![].into(),
```

Comment:
Random idea. We could respond with the next valid nonce from sequencer

Reply:
See issue #529: https://github.com/astriaorg/astria/issues/529

Reply:
Addressed by opening the issue